### PR TITLE
RD-6658 Use `cfy-agent setup` for creating an agent

### PR DESCRIPTION
**Pull request by tehasdf**
This PR is currently just a single draft WIP commit, but it'll be rewritten soon.

### CODE REVIEW

File: cloudify_agent/api/pm/base.py [hunk 1]
```diff
@@ -168,108 +161,88 @@ class Daemon(object):
...
+        logger=None,
+        broker_ip,
+        name,
+        queue,
+        deployment_id=None,
+        user=None,
+        broker_user='guest',
+        broker_pass='guest',
+        broker_vhost='/',
+        broker_ssl_enabled=False,
```
[the remark this answers is not in the document]
Comment (tehasdf) [reply cloudify_agent/api/pm/base.py:176]: They always are... .except in this CI

File: cloudify_agent/api/pm/base.py [hunk 2]
```diff
@@ -168,108 +161,88 @@ class Daemon(object):
...
-
-        # will be populated later on with runtime properties of the host
-        # node instance this agent is dedicated for (if needed)
-        self._runtime_properties = None
-
-        # configure logger
+    def __init__(
+        self,
+        *,
+        logger=None,
```
Comment (geokala) [inline cloudify_agent/api/pm/base.py:165]: Might also be worth type-hinting them?

Comment (geokala) [reply cloudify_agent/api/pm/base.py:165]: (with the validation removed that'd have some more benefit I suspect)

File: cloudify_agent/resources/script/linux.sh.template [hunk 3]
```diff
@@ -1,236 +1,273 @@
-#! /bin/bash -ex
+#!/bin/bash -e
```
Comment (geokala) [inline cloudify_agent/resources/script/linux.sh.template:1]: No point having -e here with the set -euo later

File: cloudify_agent/resources/script/linux.sh.template [hunk 4]
```diff
@@ -1,236 +1,273 @@
...
+
+    log "Agent started successfully!"
 }
-export -f main
-{% if tmpdir %}
-export TMPDIR="{{ tmpdir }}"
-{% endif %}
-main
+
+main 2>&1 | run_as_root tee "$LOG_FILE"
```
Comment (geokala) [inline cloudify_agent/resources/script/linux.sh.template:273]: I was wondering where you were doing something like this. Nice.

File: cloudify_agent/installer/script.py [hunk 5]
```diff
@@ -226,13 +168,12 @@ def install_script_download_link(cloudify_agent=None, **_):
...
+    # back-compat
     script_builder = _get_script_builder(cloudify_agent=cloudify_agent)
-    return script_builder.init_script()
+    return script_builder.install_script()
 
 
-def init_script_download_link(cloudify_agent=None, **_):
-    script_builder = _get_script_builder(cloudify_agent=cloudify_agent)
-    return script_builder.init_script_download_link()
+install_script = init_script
```
Comment (geokala) [inline cloudify_agent/installer/script.py:176]: Doesn't this entirely shadow install_script?

File: cloudify_agent/api/pm/base.py (hunk 2 above)
Comment (tehasdf) [reply cloudify_agent/api/pm/base.py:165]: eh, not that useful. It's all strings...

File: cloudify_agent/installer/script.py (hunk 5 above)
Comment (tehasdf) [reply cloudify_agent/installer/script.py:176]: we used to have both `init_script` and `install_script` here, which emitted the linux.sh/linux-download.sh - both eventually did the same thing.

I keep both functions on this level for some "back-compat" - we don't need to, but eh, why not. This is mostly only called from `ctx.plugin.init_script()` (which calls `.init_script` from here)

Anyway, shadow - not really, there's no other function named `install_script` here

File: cloudify_agent/api/pm/base.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (geokala) [reply cloudify_agent/api/pm/base.py:174]: So CI is running a test that is significantly different from general use? That's not too good. Perhaps we should stop doing that and rely on the inte/sys tests more?

File: cloudify_agent/installer/script.py (hunk 5 above)
Comment (geokala) [reply cloudify_agent/installer/script.py:176]: But you defined install_script earlier in the commit (or left it defined) but then just overrode it here- unless I misread?

Comment (tehasdf) [reply cloudify_agent/installer/script.py:176]: nah, there's the `install_script` method on the `AgentInstallationScriptBuilder`, but this is a module-level function (which eventually calls that method)

File: cloudify_agent/api/pm/base.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (tehasdf) [reply cloudify_agent/api/pm/base.py:174]: Eventually so, but I think I've already removed enough tests in this PR :)

**Review by geokala — APPROVED** [review]

**Pull request merged**